### PR TITLE
feat: 1. add toc;

### PR DESCRIPTION
**Pull request by mitsuhatu**

### CODE REVIEW

**Comment by mitsuhatu** [comment]
don't merge, I will adjust some code style soon.

File: apps/ligo-virgo/src/pages/workspace/docs/components/toc/TOC.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+            unobserve?.();
+        };
+    }, [updateTocDataSource, workspace_id]);
+
+    const onClick = async (blockId?: string) => {
+        if (blockId === activeBlockId) {
+            return;
+        }
+
+        console.log(blockId);
```
Comment (lawvs) [inline apps/ligo-virgo/src/pages/workspace/docs/components/toc/TOC.tsx:158]:
```suggestion
```

File: libs/datasource/db-service/src/services/workspace/toc.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { ServiceBaseClass } from '../base';
+import { ReturnUnobserve } from '../database/observer';
+import { ObserveCallback } from './page-tree';
+
+export class TOC extends ServiceBaseClass {
```
Comment (lawvs) [inline libs/datasource/db-service/src/services/workspace/toc.ts:5]: Is the naming styles suitable?

File: apps/ligo-virgo/src/pages/workspace/docs/utils/toc.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    return await Promise.all(
+        asyncBlocks.map(async (asyncBlock: AsyncBlock) => {
+            const asyncBlocks = await asyncBlock.children();
+
+            if (asyncBlocks?.length) {
+                return getContentByAsyncBlocks(asyncBlocks, callback);
+            }
+
+            /* get update notice */
+            asyncBlock.onUpdate(callback);
```
Comment (alt1o) [inline apps/ligo-virgo/src/pages/workspace/docs/utils/toc.ts:25]: 没有移除onUpdate的事件监听

File: apps/ligo-virgo/src/pages/workspace/docs/components/toc/TOC.tsx (hunk 1 above)
Comment (mitsuhatu) [reply apps/ligo-virgo/src/pages/workspace/docs/components/toc/TOC.tsx]: done

File: apps/ligo-virgo/src/pages/workspace/docs/utils/toc.ts (hunk 3 above)
Comment (mitsuhatu) [reply apps/ligo-virgo/src/pages/workspace/docs/utils/toc.ts]: done

**Comment by mitsuhatu** [comment]
**I find it always have a problem:** 
if add a group, there will not done update notice, because new block don't bind update handler.

so, service must notice to view when page has change.

**Comment by mitsuhatu** [comment]
branch name error.

**Pull request closed without merging**